src/app/api/activity-logs/services/activity-log.service.ts

```diff
@@ -144,4 +163,42 @@ export class ActivityLogService extends BaseService {
         return payload
     }
   }
+
+  private filterActivityLogsForLimitedAccess(
+    parsedActivityLogs: DBActivityLogArray,
+    clientUsers: ClientsResponse,
+    companies: CompaniesResponse,
+    currentInternalUser: InternalUsers,
+  ): DBActivityLogArray {
+    const previousAssigneeIds = parsedActivityLogs
+      .filter(
+        (log) =>
+          log.type === 'TASK_ASSIGNED' &&
+          (clientUsers.data?.some((client) => client.id == log.details.oldValue) ||
+            companies.data?.some((company) => company.id == log.details.oldValue)),
+      )
+      .map((log) => log.details.oldValue)
+      .reverse()
+
+    const previousUnaccessibleAssignee = previousAssigneeIds.find((id) => {
+      const otherUser =
+        clientUsers.data?.find((client) => client.id === id) || companies.data?.find((company) => company.id === id)
```

Comment:
```suggestion
      let unaccessibleUser, companyId
      
      const unaccessibleClient = clientUsers.data?.find((client) => client.id === id)
      if (unaccessibleClient) {
        unaccessibleUser = unaccessibleClient
        companyId = unaccessibleClient.companyId
      } else {
        unaccessibleUser = companies.data?.find((company) => company.id === id)
        companyId = unaccessibleUser?.id
      }
      
      if (unaccessibleUser) {
        return !currentInternalUser.companyAccessList?.includes(z.string().parse(companyId))
      }
```

Note that `unaccessibleUser` can still be null after going through the `else` block